de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/concurrent/watchdog/ConsistencyWatchdogServer.java

```diff
@@ -253,8 +253,23 @@ private void updateChecksum(SPath docPath, Set<SPath> localEditors,
             documentChecksums.put(docPath, checksum);
         }
 
-        if (!checksum.isDirty())
+        /*
+         * Ensures that the watchdog server doesn't use outdated checksums for
+         * files that no longer exist locally.
+         */
+        if (!docPath.getFile().exists()) {
+            LOG.debug("Marking checksum for " + docPath
```

Comment:
I find this log message very confusing because you will read it over and over again until the last affected user has started the recovery. You might think that marking the checksum as dirty has an actual benefit but you only have to do this because otherwise you could risk that the update(null) call might not be applied to the checksum.

Reply:
You are right. I improved the logic to check whether the content of the checksum does not match the file state so that this check is only called once. Furthermore, I changed the log message to state that the checksum is updated instead of just stating that it is marked as dirty (as marking it as dirty does not do anything in itself, as you pointed out).